File: pset 1 (C)/credit.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>


int main(void)
{
    // declare variables
    long long cc_num = 0; // intialize credit card number
    char cc_string [50];
    int checksum = 0;
    char checksum_string1[10] = {0}; // part of number to be used in step one of checksum
    char checksum_string2[10] = {0}; // part of number to be added in step two of checksum


    // gets user input
    printf("Number:");
    cc_num = get_long_long();

    // turn the cc_num into a string
    sprintf(cc_string, "%lli", cc_num);

    int n = strlen(cc_string);
    // gets the numbers to be multiplied by two for checksum
    for (int i = n-2, j = 0; i >= 0; i -= 2)
    {
        checksum_string1[j] = cc_string[i];
        j++;
    }

    // calculates step one of the checksum
    for (int i = 0, m = strlen(checksum_string1); i < m; i++)
    {
        int num = (checksum_string1[i] - '0') * 2;
        checksum +=  num / 10 + num % 10;
    }

    // gets the remaining numbers to be used in the checksum
    for (int i = n-1, j = 0; i >= 0; i -= 2)
    {
        checksum_string2[j] = cc_string[i];
        j++;
    }

    // adds remaining digites to checksum
    for (int i = 0, m = strlen(checksum_string2); i < m; i++)
    {

        checksum +=  (checksum_string2[i] - '0');
    }

    // checks checksum
    if (checksum % 10 == 0)
    {
        if (cc_string[0] == '4')
        {
            printf("VISA\n");
        }
        if (cc_string[0] == '3')
        {
            switch (cc_string[1]) {
                case '4':
                case '7':
                    printf("AMEX\n");
            }
        }
        if (cc_string[0] == '5')
        {
            switch(cc_string[1]) {
                case '1':
                case '2':
                case '3':
                case '4':
                case '5':
                    printf("MASTERCARD\n");
            }
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: pset6/credit.py
import math

def main():
    # gets user input and saves the credit card number
    cc_num = input("Number: ")

    # gets the numbers to be multiplied by two for checksum
    i = 0 # counting variable
    product_str = "" #this is the part of the check sum that *2
    add_str = "" # just add these to checksum

    if len(cc_num) % 2 == 0:
        for place in cc_num:
            if i % 2 == 0:
                product_str += place
            else:
                add_str += place
            i += 1
    else:
        for place in cc_num:
            if i % 2 == 0:
                add_str += place
            else:
                product_str += place
            i += 1

    # multiply the prouct_str by 2 and adds the digits to checksum
    checksum = 0
    for place in product_str:
        num = int(place) * 2
        checksum += math.floor(num/10) + num%10

    # adds the add_str to make final checksum
    for place in add_str:
        checksum += int(place)

    # checks checksum
    if checksum % 10 == 0:
        if cc_num[0] == '4':
            print("VISA");
        elif cc_num[0] == '3'and cc_num[1] == '4' or '7':
            print("AMEX")
        elif int(cc_num) >= 5100000000000000 and int(cc_num) < 5600000000000000:
            print("MASTERCARD")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()
